// ngx-extended-pdf-viewer/esm5/lib/toolbar/pdf-findbar/pdf-findbar-options-two-container/pdf-find-entire-phrase/pdf-find-entire-phrase.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/toolbar/pdf-findbar/pdf-findbar-options-two-container/pdf-find-entire-phrase/pdf-find-entire-phrase.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component } from '@angular/core';
import { PdfFindbarService } from '../../pdf-findbar-service';
var PdfFindMultipleSearchTextsComponent = /** @class */ (function () {
    function PdfFindMultipleSearchTextsComponent(pdfFindbarService) {
        this.pdfFindbarService = pdfFindbarService;
    }
    /**
     * @return {?}
     */
    PdfFindMultipleSearchTextsComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () { };
    PdfFindMultipleSearchTextsComponent.decorators = [
        { type: Component, args: [{
                    selector: 'pdf-find-entire-phrase',
                    template: "<input\n  type=\"checkbox\"\n  id=\"findMultipleSearchTexts\"\n  class=\"toolbarField\"\n  [(ngModel)]=\"pdfFindbarService.multipleSearchTexts\"\n/>\n\n<label\n  for=\"findMultipleSearchTexts\"\n  class=\"toolbarLabel\"\n  data-l10n-id=\"find_multiple_texts_label\">\n  multiple search texts\n</label>\n\n<input [class.hidden]=\"!pdfFindbarService.multipleSearchTexts\"\n  type=\"checkbox\"\n  id=\"individualWordsMode\"\n  class=\"toolbarField\"\n  [(ngModel)]=\"pdfFindbarService.individualWordsMode\"\n/>\n\n<label [class.hidden]=\"!pdfFindbarService.multipleSearchTexts\"\n  for=\"individualWordsMode\"\n  id=\"individualWordsModeLabel\"\n  class=\"toolbarLabel\"\n  data-l10n-id=\"find_individual_words_label\">\n  separated by word boundaries\n</label>\n",
                    styles: [""]
                }] }
    ];
    /** @nocollapse */
    PdfFindMultipleSearchTextsComponent.ctorParameters = function () { return [
        { type: PdfFindbarService }
    ]; };
    return PdfFindMultipleSearchTextsComponent;
}());
export { PdfFindMultipleSearchTextsComponent };
if (false) {
    /** @type {?} */
    PdfFindMultipleSearchTextsComponent.prototype.pdfFindbarService;
}
