// angulartics2/uiroutermodule/bundles/angulartics2-uiroutermodule.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@uirouter/core'), require('rxjs/Subject'), require('tslib'), require('angulartics2')) :
    typeof define === 'function' && define.amd ? define('angulartics2/uiroutermodule', ['exports', '@angular/core', '@uirouter/core', 'rxjs/Subject', 'tslib', 'angulartics2'], factory) :
    (factory((global.angulartics2 = global.angulartics2 || {}, global.angulartics2.uiroutermodule = {}),global.ng.core,global.core$1,global.Rx,global.tslib,global.angulartics2));
}(this, (function (exports,core,core$1,Subject,tslib,angulartics2) { 'use strict';

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    /**
     * Track Route changes for applications using UI-Router
     *
     * @link https://ui-router.github.io/ng2/docs/latest/
     *
     * referenced: https://github.com/ui-router/sample-app-angular/blob/9adb533b85c0f0fccef23968489cca0a5ec84654/src/app/util/ga.ts
     */
    var UIRouterTracking = (function () {
        /**
         * @param {?} transitionService
         */
        function UIRouterTracking(transitionService) {
            this.transitionService = transitionService;
        }
        /**
         * @param {?} trans
         * @return {?}
         */
        UIRouterTracking.prototype.path = function (trans) {
            return trans.$to().url.format(trans.params());
        };
        /**
         * @param {?} settings
         * @return {?}
         */
        UIRouterTracking.prototype.trackLocation = function (settings) {
            var _this = this;
            var /** @type {?} */ subject = new Subject.Subject();
            this.transitionService.onSuccess({}, function (trans) {
                return subject.next({ url: _this.path(trans) });
            }, {
                priority: -10000,
            });
            return subject;
        };
        /**
         * @param {?} url
         * @return {?}
         */
        UIRouterTracking.prototype.prepareExternalUrl = function (url) {
            return url;
        };
        return UIRouterTracking;
    }());
    UIRouterTracking.decorators = [
        { type: core.Injectable },
    ];
    /** @nocollapse */
    UIRouterTracking.ctorParameters = function () {
        return [
            { type: core$1.TransitionService, },
        ];
    };
    var Angulartics2UirouterModule = (function () {
        function Angulartics2UirouterModule() {
        }
        /**
         * @param {?} providers
         * @param {?=} settings
         * @return {?}
         */
        Angulartics2UirouterModule.forRoot = function (providers, settings) {
            if (settings === void 0) {
                settings = {};
            }
            return {
                ngModule: Angulartics2UirouterModule,
                providers: tslib.__spread([
                    { provide: angulartics2.ANGULARTICS2_TOKEN, useValue: { providers: providers, settings: settings } },
                    angulartics2.Angulartics2,
                    { provide: angulartics2.RouterlessTracking, useClass: UIRouterTracking }
                ], providers),
            };
        };
        return Angulartics2UirouterModule;
    }());
    Angulartics2UirouterModule.decorators = [
        { type: core.NgModule, args: [{
                    imports: [angulartics2.Angulartics2OnModule],
                },] },
    ];

    exports.Angulartics2UirouterModule = Angulartics2UirouterModule;
    exports.UIRouterTracking = UIRouterTracking;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
