// angulartics2/uiroutermodule/esm2015/uirouter.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { NgModule } from '@angular/core';
import { Angulartics2, Angulartics2OnModule, ANGULARTICS2_TOKEN, RouterlessTracking, } from 'angulartics2';
import { UIRouterTracking } from './uirouter';
export class Angulartics2UirouterModule {
    /**
     * @param {?} providers
     * @param {?=} settings
     * @return {?}
     */
    static forRoot(providers, settings = {}) {
        return {
            ngModule: Angulartics2UirouterModule,
            providers: [
                { provide: ANGULARTICS2_TOKEN, useValue: { providers, settings } },
                Angulartics2,
                { provide: RouterlessTracking, useClass: UIRouterTracking },
                ...providers,
            ],
        };
    }
}
Angulartics2UirouterModule.decorators = [
    { type: NgModule, args: [{
                imports: [Angulartics2OnModule],
            },] },
];
function Angulartics2UirouterModule_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    Angulartics2UirouterModule.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    Angulartics2UirouterModule.ctorParameters;
}
