// angulartics2/esm2015/angulartics2-interfaces.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/**
 * @record
 */
export function EventTrack() { }
function EventTrack_tsickle_Closure_declarations() {
    /** @type {?} */
    EventTrack.prototype.action;
    /** @type {?} */
    EventTrack.prototype.properties;
}
/**
 * @record
 */
export function PageTrack() { }
function PageTrack_tsickle_Closure_declarations() {
    /** @type {?} */
    PageTrack.prototype.path;
}
/**
 * @link https://developers.google.com/analytics/devguides/collection/analyticsjs/user-timings
 * @record
 */
export function UserTimings() { }
function UserTimings_tsickle_Closure_declarations() {
    /**
     * A string for categorizing all user timing variables into logical groups (e.g. 'JS Dependencies').
     * @type {?}
     */
    UserTimings.prototype.timingCategory;
    /**
     * A string to identify the variable being recorded (e.g. 'load').
     * @type {?}
     */
    UserTimings.prototype.timingVar;
    /**
     * The number of milliseconds in elapsed time to report to Google Analytics (e.g. 20).
     * @type {?}
     */
    UserTimings.prototype.timingValue;
    /**
     * A string that can be used to add flexibility in visualizing user timings in the reports (e.g. 'Google CDN').
     * @type {?|undefined}
     */
    UserTimings.prototype.timingLabel;
}
