// ngx-extended-pdf-viewer/esm2015/lib/toolbar/pdf-toolbar/pdf-toolbar.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/toolbar/pdf-toolbar/pdf-toolbar.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input, TemplateRef } from '@angular/core';
export class PdfToolbarComponent {
    constructor() {
        this.mobileFriendlyZoomScale = 1;
        this.primaryMenuVisible = true;
        this.showBookmarkButton = true;
        this.showDownloadButton = true;
        this.showFindButton = undefined;
        this.showHandToolButton = true;
        this.showOpenFileButton = true;
        this.showPrintButton = true;
        this.showPagingButtons = true;
        this.showPresentationModeButton = false;
        this.showRotateButton = true;
        this.showSecondaryToolbarButton = true;
        this.showSidebarButton = true;
        this.showZoomButtons = true;
        this.textLayer = undefined;
        this.toolbarMarginTop = '0px';
        this.toolbarWidth = '100%';
        this.zoomLevels = ['auto', 'page-actual', 'page-fit', 'page-width', 0.5, 0.75, 1, 1.25, 1.5, 2, 3, 4];
    }
}
PdfToolbarComponent.decorators = [
    { type: Component, args: [{
                selector: 'pdf-toolbar',
                template: "<div class=\"toolbar\" [class.invisible]=\"!primaryMenuVisible\"\n    [style.transform]=\"'scale(' + mobileFriendlyZoomScale + ')'\"\n    [style.transformOrigin]=\"'left center'\"\n    [style.width]=\"toolbarWidth\"\n    [style.marginTop]=\"toolbarMarginTop\"\n>\n  <div id=\"toolbarContainer\">\n    <ng-content *ngTemplateOutlet=\"customToolbar ? customToolbar : defaultToolbar\"></ng-content>\n    <div id=\"loadingBar\">\n      <div class=\"progress\"><div class=\"glimmer\"></div></div>\n    </div>\n  </div>\n</div>\n\n<ng-template #defaultToolbar>\n  <div id=\"toolbarViewer\">\n    <div id=\"toolbarViewerLeft\">\n      <pdf-toggle-sidebar [showSidebarButton]=\"showSidebarButton\"></pdf-toggle-sidebar>\n      <pdf-find-button [showFindButton]=\"showFindButton\" [textLayer]=\"textLayer\"></pdf-find-button>\n      <pdf-paging-area [showPagingButtons]=\"showPagingButtons\"></pdf-paging-area>\n    </div>\n    <pdf-zoom-toolbar [showZoomButtons]=\"showZoomButtons\" [zoomLevels]=\"zoomLevels\"></pdf-zoom-toolbar>\n    <div id=\"toolbarViewerRight\">\n      <pdf-hand-tool [showHandToolButton]=\"showHandToolButton\"></pdf-hand-tool>\n      <pdf-select-tool [showSelectToolButton]=\"showHandToolButton\"></pdf-select-tool>\n      <pdf-rotate-page [showRotateButton]=\"showRotateButton\" [clockwise]=\"true\" [counterClockwise]=\"false\"></pdf-rotate-page>\n      <pdf-rotate-page [showRotateButton]=\"showRotateButton\" [clockwise]=\"false\" [counterClockwise]=\"true\"></pdf-rotate-page>\n      <pdf-presentation-mode [showPresentationModeButton]=\"showPresentationModeButton\"></pdf-presentation-mode>\n      <pdf-open-file [showOpenFileButton]=\"showOpenFileButton\"></pdf-open-file>\n      <pdf-print [showPrintButton]=\"showPrintButton\"></pdf-print>\n      <pdf-download [showDownloadButton]=\"showDownloadButton\"></pdf-download>\n      <pdf-bookmark [showBookmarkButton]=\"showBookmarkButton\"></pdf-bookmark>\n      <pdf-toggle-secondary-toolbar [showSecondaryToolbarButton]=\"showSecondaryToolbarButton\"></pdf-toggle-secondary-toolbar>\n    </div>\n   </div>\n</ng-template>\n",
                styles: [""]
            }] }
];
/** @nocollapse */
PdfToolbarComponent.ctorParameters = () => [];
PdfToolbarComponent.propDecorators = {
    customToolbar: [{ type: Input }],
    mobileFriendlyZoomScale: [{ type: Input }],
    primaryMenuVisible: [{ type: Input }],
    showBookmarkButton: [{ type: Input }],
    showDownloadButton: [{ type: Input }],
    showFindButton: [{ type: Input }],
    showHandToolButton: [{ type: Input }],
    showOpenFileButton: [{ type: Input }],
    showPrintButton: [{ type: Input }],
    showPagingButtons: [{ type: Input }],
    showPresentationModeButton: [{ type: Input }],
    showRotateButton: [{ type: Input }],
    showSecondaryToolbarButton: [{ type: Input }],
    showSidebarButton: [{ type: Input }],
    showZoomButtons: [{ type: Input }],
    textLayer: [{ type: Input }],
    toolbarMarginTop: [{ type: Input }],
    toolbarWidth: [{ type: Input }],
    zoomLevels: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    PdfToolbarComponent.prototype.customToolbar;
    /** @type {?} */
    PdfToolbarComponent.prototype.mobileFriendlyZoomScale;
    /** @type {?} */
    PdfToolbarComponent.prototype.primaryMenuVisible;
    /** @type {?} */
    PdfToolbarComponent.prototype.showBookmarkButton;
    /** @type {?} */
    PdfToolbarComponent.prototype.showDownloadButton;
    /** @type {?} */
    PdfToolbarComponent.prototype.showFindButton;
    /** @type {?} */
    PdfToolbarComponent.prototype.showHandToolButton;
    /** @type {?} */
    PdfToolbarComponent.prototype.showOpenFileButton;
    /** @type {?} */
    PdfToolbarComponent.prototype.showPrintButton;
    /** @type {?} */
    PdfToolbarComponent.prototype.showPagingButtons;
    /** @type {?} */
    PdfToolbarComponent.prototype.showPresentationModeButton;
    /** @type {?} */
    PdfToolbarComponent.prototype.showRotateButton;
    /** @type {?} */
    PdfToolbarComponent.prototype.showSecondaryToolbarButton;
    /** @type {?} */
    PdfToolbarComponent.prototype.showSidebarButton;
    /** @type {?} */
    PdfToolbarComponent.prototype.showZoomButtons;
    /** @type {?} */
    PdfToolbarComponent.prototype.textLayer;
    /** @type {?} */
    PdfToolbarComponent.prototype.toolbarMarginTop;
    /** @type {?} */
    PdfToolbarComponent.prototype.toolbarWidth;
    /** @type {?} */
    PdfToolbarComponent.prototype.zoomLevels;
}
