// ngx-extended-pdf-viewer/esm5/lib/toolbar/pdf-paging-area/pdf-page-number/pdf-page-number.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/toolbar/pdf-paging-area/pdf-page-number/pdf-page-number.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
var PdfPageNumberComponent = /** @class */ (function () {
    function PdfPageNumberComponent() {
        this.showPagingButtons = true;
    }
    /**
     * @return {?}
     */
    PdfPageNumberComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () { };
    PdfPageNumberComponent.decorators = [
        { type: Component, args: [{
                    selector: 'pdf-page-number',
                    template: "<input\n  [class.invisible]=\"!showPagingButtons\"\n  type=\"number\"\n  id=\"pageNumber\"\n  class=\"toolbarField pageNumber\"\n  title=\"Page\"\n  value=\"1\"\n  size=\"4\"\n  min=\"1\"\n  data-l10n-id=\"page\"\n/>\n<span [class.invisible]=\"!showPagingButtons\" id=\"numPages\" class=\"toolbarLabel hiddenXLView\"></span>\n",
                    styles: [""]
                }] }
    ];
    /** @nocollapse */
    PdfPageNumberComponent.ctorParameters = function () { return []; };
    PdfPageNumberComponent.propDecorators = {
        showPagingButtons: [{ type: Input }]
    };
    return PdfPageNumberComponent;
}());
export { PdfPageNumberComponent };
if (false) {
    /** @type {?} */
    PdfPageNumberComponent.prototype.showPagingButtons;
}
