// ngx-extended-pdf-viewer/esm5/lib/toolbar/pdf-rotate-page/pdf-rotate-page.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/toolbar/pdf-rotate-page/pdf-rotate-page.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { PDFNotificationService } from './../../pdf-notification-service';
import { Component, Input, ViewChild, ElementRef } from '@angular/core';
var PdfRotatePageComponent = /** @class */ (function () {
    function PdfRotatePageComponent(notificationService) {
        var _this = this;
        this.notificationService = notificationService;
        this.showRotateButton = true;
        this.disableRotate = true;
        this.clockwise = true;
        this.counterClockwise = true;
        /** @type {?} */
        var subscription = this.notificationService.onPDFJSInit.subscribe((/**
         * @return {?}
         */
        function () {
            _this.onPdfJsInit();
            subscription.unsubscribe();
        }));
    }
    /**
     * @return {?}
     */
    PdfRotatePageComponent.prototype.rotateCW = /**
     * @return {?}
     */
    function () {
        /** @type {?} */
        var PDFViewerApplication = ((/** @type {?} */ (window))).PDFViewerApplication;
        PDFViewerApplication.eventBus.dispatch('rotatecw');
    };
    /**
     * @return {?}
     */
    PdfRotatePageComponent.prototype.rotateCCW = /**
     * @return {?}
     */
    function () {
        /** @type {?} */
        var PDFViewerApplication = ((/** @type {?} */ (window))).PDFViewerApplication;
        PDFViewerApplication.eventBus.dispatch('rotateccw');
    };
    /**
     * @return {?}
     */
    PdfRotatePageComponent.prototype.onPdfJsInit = /**
     * @return {?}
     */
    function () {
        var _this = this;
        /** @type {?} */
        var PDFViewerApplication = ((/** @type {?} */ (window))).PDFViewerApplication;
        PDFViewerApplication.eventBus.on('updateuistate', (/**
         * @param {?} event
         * @return {?}
         */
        function (event) { return _this.updateUIState(event); }));
    };
    /**
     * @param {?} event
     * @return {?}
     */
    PdfRotatePageComponent.prototype.updateUIState = /**
     * @param {?} event
     * @return {?}
     */
    function (event) {
        this.disableRotate = event.pagesCount === 0;
        if (this.button1) {
            this.button1.nativeElement.disabled = this.disableRotate;
        }
        if (this.button2) {
            this.button2.nativeElement.disabled = this.disableRotate;
        }
    };
    PdfRotatePageComponent.decorators = [
        { type: Component, args: [{
                    selector: 'pdf-rotate-page',
                    template: "<button\n  *ngIf=\"clockwise\"\n  (click)=\"rotateCW()\"\n  type=\"button\"\n  [class.invisible]=\"!showRotateButton\"\n  id=\"primaryPageRotateCw\"\n  class=\"toolbarButton hiddenXLView rotateCw\"\n  title=\"Rotate Clockwise\"\n  data-l10n-id=\"page_rotate_cw\"\n  [disabled]=\"disableRotate\"\n  #button1>\n    <svg style=\"width:23px;height:23px\" viewBox=\"0 0 24 24\">\n      <path fill=\"currentColor\" d=\"M12 3C7.03 3 3 7.03 3 12S7.03 21 12 21C14 21 15.92 20.34 17.5 19.14L16.06 17.7C14.87 18.54 13.45 19 12 19C8.13 19 5 15.87 5 12S8.13 5 12 5 19 8.13 19 12H16L20 16L24 12H21C21 7.03 16.97 3 12 3\" />\n    </svg>\n</button>\n<button\n  *ngIf=\"counterClockwise\"\n  (click)=\"rotateCCW()\"\n  type=\"button\"\n  [class.invisible]=\"!showRotateButton\"\n  id=\"primaryPageRotateCcw\"\n  class=\"toolbarButton hiddenXLView rotateCcw\"\n  title=\"Rotate Counterclockwise\"\n  data-l10n-id=\"page_rotate_ccw\"\n  [disabled]=\"disableRotate\"\n  #button2>\n    <svg style=\"width:23px;height:23px\" viewBox=\"0 0 24 24\">\n      <path fill=\"currentColor\" d=\"M12 3C7.03 3 3 7.03 3 12H0L4 16L8 12H5C5 8.13 8.13 5 12 5S19 8.13 19 12 15.87 19 12 19C10.55 19 9.13 18.54 7.94 17.7L6.5 19.14C8.08 20.34 10 21 12 21C16.97 21 21 16.97 21 12S16.97 3 12 3\" />\n    </svg>\n</button>\n",
                    styles: [":host{margin-top:-1px}:host:focus{outline:0}button:focus,svg:focus{outline:0}"]
                }] }
    ];
    /** @nocollapse */
    PdfRotatePageComponent.ctorParameters = function () { return [
        { type: PDFNotificationService }
    ]; };
    PdfRotatePageComponent.propDecorators = {
        showRotateButton: [{ type: Input }],
        clockwise: [{ type: Input }],
        counterClockwise: [{ type: Input }],
        button1: [{ type: ViewChild, args: ['button1',] }],
        button2: [{ type: ViewChild, args: ['button2',] }]
    };
    return PdfRotatePageComponent;
}());
export { PdfRotatePageComponent };
if (false) {
    /** @type {?} */
    PdfRotatePageComponent.prototype.showRotateButton;
    /** @type {?} */
    PdfRotatePageComponent.prototype.disableRotate;
    /** @type {?} */
    PdfRotatePageComponent.prototype.clockwise;
    /** @type {?} */
    PdfRotatePageComponent.prototype.counterClockwise;
    /**
     * @type {?}
     * @private
     */
    PdfRotatePageComponent.prototype.button1;
    /**
     * @type {?}
     * @private
     */
    PdfRotatePageComponent.prototype.button2;
    /**
     * @type {?}
     * @private
     */
    PdfRotatePageComponent.prototype.notificationService;
}
