// ngx-extended-pdf-viewer/esm5/lib/toolbar/pdf-toggle-secondary-toolbar/pdf-toggle-secondary-toolbar.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/toolbar/pdf-toggle-secondary-toolbar/pdf-toggle-secondary-toolbar.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
var PdfToggleSecondaryToolbarComponent = /** @class */ (function () {
    function PdfToggleSecondaryToolbarComponent() {
        this.showSecondaryToolbarButton = true;
    }
    /**
     * @return {?}
     */
    PdfToggleSecondaryToolbarComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () { };
    PdfToggleSecondaryToolbarComponent.decorators = [
        { type: Component, args: [{
                    selector: 'pdf-toggle-secondary-toolbar',
                    template: "<button\n  type=\"button\"\n  [class.invisible]=\"!showSecondaryToolbarButton\"\n  id=\"secondaryToolbarToggle\"\n  class=\"toolbarButton\"\n  title=\"Tools\"\n  data-l10n-id=\"tools\"\n>\n  <svg style=\"width:27px;height:27px\" viewBox=\"0 0 24 24\" onclick=\"event.preventDefault(); return false;\">\n    <path fill=\"currentColor\" d=\"M3,6H21V8H3V6M3,11H21V13H3V11M3,16H21V18H3V16Z\" />\n  </svg>\n  <span data-l10n-id=\"tools_label\">Tools</span>\n</button>\n",
                    styles: ["svg{margin-top:-3px}:host:focus{outline:0}button:focus,svg:focus{outline:0}"]
                }] }
    ];
    /** @nocollapse */
    PdfToggleSecondaryToolbarComponent.ctorParameters = function () { return []; };
    PdfToggleSecondaryToolbarComponent.propDecorators = {
        showSecondaryToolbarButton: [{ type: Input }]
    };
    return PdfToggleSecondaryToolbarComponent;
}());
export { PdfToggleSecondaryToolbarComponent };
if (false) {
    /** @type {?} */
    PdfToggleSecondaryToolbarComponent.prototype.showSecondaryToolbarButton;
}
