// ngx-extended-pdf-viewer/esm5/lib/toolbar/pdf-toggle-sidebar/pdf-toggle-sidebar.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/toolbar/pdf-toggle-sidebar/pdf-toggle-sidebar.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
var PdfToggleSidebarComponent = /** @class */ (function () {
    function PdfToggleSidebarComponent() {
        this.showSidebarButton = true;
    }
    /**
     * @return {?}
     */
    PdfToggleSidebarComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () { };
    PdfToggleSidebarComponent.decorators = [
        { type: Component, args: [{
                    selector: 'pdf-toggle-sidebar',
                    template: "<button\n  type=\"button\"\n  [class.invisible]=\"!showSidebarButton\"\n  id=\"sidebarToggle\"\n  title=\"Toggle Sidebar\"\n  class=\"toolbarButton\"\n  data-l10n-id=\"toggle_sidebar\"\n>\n  <svg style=\"width:24px;height:24px\" viewBox=\"0 0 24 24\">\n    <path fill=\"currentColor\" d=\"M3,9H17V7H3V9M3,13H17V11H3V13M3,17H17V15H3V17M19,17H21V15H19V17M19,7V9H21V7H19M19,13H21V11H19V13Z\" />\n  </svg>\n  <span data-l10n-id=\"toggle_sidebar_label\">Toggle Sidebar</span>\n</button>\n",
                    styles: ["button svg{margin-top:-2px}:host:focus{outline:0}button:focus,svg:focus{outline:0}"]
                }] }
    ];
    /** @nocollapse */
    PdfToggleSidebarComponent.ctorParameters = function () { return []; };
    PdfToggleSidebarComponent.propDecorators = {
        showSidebarButton: [{ type: Input }]
    };
    return PdfToggleSidebarComponent;
}());
export { PdfToggleSidebarComponent };
if (false) {
    /** @type {?} */
    PdfToggleSidebarComponent.prototype.showSidebarButton;
}
