// angulartics2/woopra/bundles/angulartics2-woopra.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('angulartics2')) :
    typeof define === 'function' && define.amd ? define('angulartics2/woopra', ['exports', '@angular/core', 'angulartics2'], factory) :
    (factory((global.angulartics2 = global.angulartics2 || {}, global.angulartics2.woopra = {}),global.ng.core,global.angulartics2));
}(this, (function (exports,core,angulartics2) { 'use strict';

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    var Angulartics2Woopra = (function () {
        /**
         * @param {?} angulartics2
         */
        function Angulartics2Woopra(angulartics2$$1) {
            var _this = this;
            this.angulartics2 = angulartics2$$1;
            if (typeof (woopra) === 'undefined') {
                console.warn('Woopra not found');
            }
            this.angulartics2.pageTrack
                .pipe(this.angulartics2.filterDeveloperMode())
                .subscribe(function (x) { return _this.pageTrack(x.path); });
            this.angulartics2.eventTrack
                .pipe(this.angulartics2.filterDeveloperMode())
                .subscribe(function (x) { return _this.eventTrack(x.action, x.properties); });
            this.angulartics2.setUserProperties
                .subscribe(function (x) { return _this.setUserProperties(x); });
        }
        /**
         * @param {?} path
         * @return {?}
         */
        Angulartics2Woopra.prototype.pageTrack = function (path) {
            try {
                woopra.track('pv', {
                    url: path
                });
            }
            catch (e) {
                if (!(e instanceof ReferenceError)) {
                    throw e;
                }
            }
        };
        /**
         * @param {?} action
         * @param {?} properties
         * @return {?}
         */
        Angulartics2Woopra.prototype.eventTrack = function (action, properties) {
            try {
                woopra.track(action, properties);
            }
            catch (e) {
                if (!(e instanceof ReferenceError)) {
                    throw e;
                }
            }
        };
        /**
         * @param {?} properties
         * @return {?}
         */
        Angulartics2Woopra.prototype.setUserProperties = function (properties) {
            try {
                if (properties.email) {
                    woopra.identify(properties);
                }
            }
            catch (e) {
                if (!(e instanceof ReferenceError)) {
                    throw e;
                }
            }
        };
        return Angulartics2Woopra;
    }());
    Angulartics2Woopra.decorators = [
        { type: core.Injectable },
    ];
    /** @nocollapse */
    Angulartics2Woopra.ctorParameters = function () {
        return [
            { type: angulartics2.Angulartics2, },
        ];
    };

    exports.Angulartics2Woopra = Angulartics2Woopra;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
